app/models/concerns/account_finder_concern.rb

```diff
@@ -25,8 +25,21 @@ def find_local(username)
     def find_remote(username, domain)
       AccountFinder.new(username, domain).account
     end
+
+    # sigh
+    def find_local_or_remote(username, domain)
+      TagManager.instance.local_domain?(domain) ? find_local(username) : find_remote(username, domain)
+    end
+
+    def validate_account_string(account_string)
+      match = ACCOUNT_STRING_RE.match(account_string)
+
+      [match[:username], match[:domain]] if match
+    end
   end
 
+  ACCOUNT_STRING_RE = /^@?(?<username>#{Account::USERNAME_RE})(?:@(?<domain>[[:word:]\.\-]+))?$/i
```

Comment:
`[:word:]` includes characters outside of the ASCII range, I’m pretty sure. If Mastodon allows IDNs elsewhere then that’s fine, but I’m asking because I’m not sure. If it doesn’t, use `\w`, which is just ASCII.

Reply:
I’d also like to move this RegExp somewhere else eventually, but I don‘t think that blocks this MR.

Reply:
i literally copypasted `MENTION_RE` (pre-#39) and added anchors to the sides

Reply:
okay cool lol; it’d be nice if we could just have one regexp instead of two but uhhh i know grouping can be a pain with that sometimes

anyway that wouldn’t block me from merging this

Reply:
yeah i would have defined it in terms of MENTION_RE if i knew a way to

Reply:
> i literally copypasted `MENTION_RE` (pre-#39) and added anchors to the sides and named matching

oh actually i made a mistake: actual `MENTION_RE` uses `[[:word:]\.\-]+[[:word]]+' for the `domain` part. so currently a `from:@a@a` would run a search, whereas if we actually duplicated mention processing behavior (which i indeded and would still want), we wouldn&#x2019;t match on `@a@a` and return a (still presently unhelpful) error